Makefile

```diff
@@ -37,6 +37,12 @@ BULK_MEMORY_THRESHOLD ?= 32
 TARGET_TRIPLE = wasm32-wasi
 MULTIARCH_TRIPLE = wasm32-wasi
 
+# Threaded version necessitates a different traget, as objects from different
+# targets can't be mixed together while linking
+ifeq ($(THREAD_MODEL), posix)
+MULTIARCH_TRIPLE = wasm32-wasi-pthread
```

Comment:
I wonder if its work being explicit here and calling this `wasm32-unknown-wasi-pthread` ?

Reply:
Also, is there a reason with have TARGET_TRIPLE and MULTIARCH_TRIPLE separated here?  I would that thought that we want to just use a single triple everywhere?

Reply:
I propose we handle the vendor string in the same way as the non-pthread builds, so that the only difference is the suffix.

Reply:
> Also, is there a reason with have TARGET_TRIPLE and MULTIARCH_TRIPLE separated here? I would that thought that we want to just use a single triple everywhere?

Should we update both to include pthreads or should we only keep one variable? Both are viable solutions I think.

Reply:
Maybe I misread your question initially. I don't know why there are two variables, but can try to make it into just one.

Reply:
I don't know the precise state of things here, but my suggestion was just to avoid making the `unknown` vendor field explicit here, since it's implicit everywhere else and making it explicit here would make more things different.

Reply:
Oh, I meant @sbc100's comment right before yours - it looks like he was asking why there are two variables.